modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/utils/imagePropsTransformer.ts

```diff
@@ -3,21 +3,82 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-export default function imagePropsTransformer(
-	imageData: any | string
-): React.ImgHTMLAttributes<HTMLImageElement> | string | undefined {
-	if (typeof imageData === 'string') {
-		return imageData;
-	}
+import getValueFromItem from '../../utils/getValueFromItem';
+
+interface IDocsAndMediaImageProps {
+	id: number;
+	link: {
+		href: string;
+		label: string;
+	};
+	name: string;
+}
 
-	if (typeof imageData === 'object') {
+interface IImageProps {
+	itemData?: any;
+	name?: string;
+	options?: {
+		label?: string;
+		labelKey?: string;
+		shape?: 'circle' | 'rounded' | 'user-icon';
+		size?: 'lg' | 'sm' | 'xl';
+	};
+	value?:
+		| string
+		| IDocsAndMediaImageProps
+		| React.ImgHTMLAttributes<HTMLImageElement>;
+}
 
-		// Docs & media object
+export default function imagePropsTransformer(
+	imageData: IDocsAndMediaImageProps | IImageProps | string | undefined
+): React.ImgHTMLAttributes<HTMLImageElement> | undefined {
+	let imageProps: React.ImgHTMLAttributes<HTMLImageElement> = {
+		alt: '',
+		src: undefined,
+	};
 
-		return {
-			alt: imageData?.link.label,
-			id: imageData?.id,
-			src: imageData?.link.href,
+	if (!imageData || !Object.keys(imageData).length) {
+		return undefined;
```

Comment:
[`CardWithInfo`](https://github.com/liferay/clay/blob/master/packages/clay-card/src/CardWithInfo.tsx#L141) uses this `undefined` to provide a default background icon.